File: Lua/Fibonacci.lua
function fib(n)
  function loop(n1, n2, count)
    if count == n then return n2
    else
      return loop(n2, n1+n2, count+1)
    end
  end

  if n == 0 then
    return 0
  else
    return loop(0, 1, 1)
  end
end

print(fib(0))
print(fib(1))
print(fib(2))
print(fib(3))
print(fib(4))
print(fib(5))

--[[
for line in io.lines(arg[1]) do
    io.write(fib(tonumber(line)) .. '\n')
end
--]]

File: Python3/Fibonacci.py

def fib(n: int):
    n1, n2, count = 0, 0, 0
    while True:
        if count == n: return n2
        else:
            if count == 0:
                n1, n2, count = 0, 1, 1
            else:
                n1, n2, count = n2, (n1 + n2), (count + 1)

print(fib(22))

x = "42"
int(x)
